x/posts/internal/types/post.go

```diff
@@ -1,77 +1,51 @@
 package types
 
 import (
+	"crypto/sha256"
+	"encoding/hex"
 	"encoding/json"
 	"fmt"
 	"strconv"
 	"strings"
 	"time"
 	"unicode"
 
-	"github.com/desmos-labs/desmos/x/commons"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/commons"
 )
 
 // ---------------
 // --- Post id
 // ---------------
 
 // PostID represents a unique post id
-type PostID uint64
+type PostID string
 
-// Valid tells if the id can be used safely
-func (id PostID) Valid() bool {
-	return id != 0
+// ComputeID returns a sha256 hash of the given data concatenated together
+// nolint: interfacer
+func ComputeID(creationDate time.Time, creator sdk.AccAddress, subspace string) PostID {
+	hash := sha256.Sum256([]byte(creationDate.String() + creator.String() + subspace))
+	return PostID(hex.EncodeToString(hash[:]))
 }
 
-// Next returns the subsequent id to this one
-func (id PostID) Next() PostID {
-	return id + 1
+// Valid tells if the id can be used safely
+func (id PostID) Valid() error {
```

Comment:
I'm really not a big fan of this method returning an `error`. Since the error can only be one (the post id is not valid), it can be much easier be represented by a `bool` return value instead.

Can we switch it back to it returning `bool` instead of `err`? This should also be much more easier to use.

Reply:
I've switch to this cause there's a lot of repetitions of the same error string, so I think it would be better doing it like this..